### PR TITLE
Private harbor fix

### PR DESCRIPTION
**Pull request by at88mph**

### CODE REVIEW

**Comment by at88mph** [comment]
@szautkin Please look at the JavaScript changes.  The Java changes are more optional.

File: src/react/SciencePortalPrivateForm.js [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+                </Col>
+                <Col sm={7}>
+                  <Form.Select
+                      value={this.state.fData.selectedType}
+                      onChange={this.state.fData.changeTypeHandler}
+                      name="type"
+                      size="sm"
+                      className="sp-form-cursor"
+                  >
+                    {this.state.fData.types.map(mapObj => (
```
Comment (szautkin) [inline src/react/SciencePortalPrivateForm.js]: I would suggest to use optional chaining `?.` to guard against possible undefined/null objects

File: src/react/SciencePortalPrivateForm.js [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+                        {this.renderPopover("Memory", "System memory (RAM) in gigabytes.")}
+                      </Form.Label>
+                    </Col>
+                    <Col sm={7}>
+                      <Form.Select
+                          value={this.state.selectedRAM || this.state.fData.contextData.defaultRAM}
+                          name="ram"
+                          className="sp-form-cursor"
+                          onChange={this.handleRAMChange.bind(this)}>
+                        {this.state.fData.contextData.availableRAM.map(mapObj => (
```
Comment (szautkin) [inline src/react/SciencePortalPrivateForm.js]: same here

Comment (szautkin) [reply src/react/SciencePortalPrivateForm.js]: ?.map
Screenshot 2024-11-06 at 5 52 03 PM

File: src/react/SciencePortalPrivateForm.js (hunk 1 above)
Comment (at88mph) [reply src/react/SciencePortalPrivateForm.js]: Thanks, Done!

File: src/react/SciencePortalPrivateForm.js (hunk 2 above)
Comment (at88mph) [reply src/react/SciencePortalPrivateForm.js]: That's a good idea, thanks.  Done.

**Review by szautkin — DISMISSED** [review]

**Review by jburke-cadc — APPROVED** [review]

**Review by szautkin — APPROVED** [review]

**Pull request merged**